url47ebb27197_asirbu_CorrectiveEvolution_tgz-pJ8-eu_fbk_soa_evolution_engine_impl_test_ProblemToSMVFuncTestJ8/src/eu/fbk/soa/evolution/sts/impl/NusmvState.java

```diff
@@ -0,0 +1,105 @@
+package eu.fbk.soa.evolution.sts.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import eu.fbk.soa.evolution.sts.State;
+
+
+
+public class NusmvState implements State {
+	
+	private static int nrStates = 1;
+	
+	private int index;
+	
+	private Map<String, String> variableValues;
+	
+	private String name;
+	
+	public NusmvState() {
+		index = nrStates;
+		name = "s"+index;
+		nrStates++;
+		variableValues = new HashMap<String, String>(); 
+	}
+	
+	public NusmvState(Map<String, String> varValues) {		
+		this();
+		for (String key : varValues.keySet()) {
+			if (key.contains(".state") || key.contains(".State")) {
+				String value = varValues.get(key);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.